components/button/tests/unit/button.spec.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { Button } from '@westpac/button';
+import { GEL } from '@westpac/core';
+import { fireEvent, render, screen, within } from '@testing-library/react';
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import wbc from '@westpac/wbc';
+import { ArrowLeftIcon } from '@westpac/icon';
+
+// The default tests every component should run
+overridesTest({
+	name: 'button', // the name has to be the package name without '@westpac/' scope
+	overrides: ['Button'], // every single override root key
+	Component: (props) => <Button {...props}>Button content</Button>, // the component with all components rendered
+});
+
+// another default test to check that the component errors when outside of GEL and renders when inside
+nestingTest({
+	name: 'button',
+	Component: (props) => <Button {...props} />,
+});
+
+const SimpleButton = (props) => (
+	<GEL brand={wbc}>
+		<Button {...props}>{props.children}</Button>
+	</GEL>
+);
+
+// Component specific tests
+describe('Button component', () => {
+	const originalLog = console.log;
```

Comment:
Instead of overriding the console.log, create a simple React component which uses the button component. Then within the component you could change text or equal to see if the on click is working.

Reply:
Please see my changes in the following commit: 02544070
I left the test that makes a console.log and then spies on it - I don't think this one actually overrides console.log, like the lines I removed did. But I'll remove the test if you want it gone.